packages/typespec-go/src/tcgcadapter/types.ts

```diff
@@ -572,43 +523,7 @@ export class typeAdapter {
       // property is on a model that's not the root discriminator
       annotations.isDiscriminator = true;
       field.defaultValue = this.getDiscriminatorLiteral(prop);
-    } /*else if (prop.clientDefaultValue) {
-      if (!go.isLiteralValueType(field.type)) {
-        throw new Error(`unsupported default value type ${go.getTypeDeclaration(field.type)} for field ${field.fieldName}`);
-      }
-      if (go.isConstantType(field.type)) {
-        // find the corresponding ConstantValue
-        const constType = types.get(field.type.name);
-        if (!constType) {
-          throw new Error(`didn't find ConstantType for ${field.type.name}`);
-        }
-        let found = false;
-        for (const val of values((<go.ConstantType>constType).values)) {
-          if (val.value === prop.clientDefaultValue) {
-            const keyName = `literal-${val.valueName}`;
-            let literalValue = types.get(keyName);
-            if (!literalValue) {
-              literalValue = new go.LiteralValue(field.type, val);
-              types.set(keyName, literalValue);
-            }
-            field.defaultValue = <go.LiteralValue>literalValue;
-            found = true;
-            break;
-          }
-        }
-        if (!found) {
-          throw new Error(`didn't find ConstantValue for ${prop.clientDefaultValue}`);
-        }
-      } else {
-        const keyName = `literal-${go.getTypeDeclaration(field.type)}-${prop.clientDefaultValue}`;
-        let literalValue = types.get(keyName);
-        if (!literalValue) {
-          literalValue = new go.LiteralValue(field.type, prop.clientDefaultValue);
-          types.set(keyName, literalValue);
-        }
-        field.defaultValue = <go.LiteralValue>literalValue;
-      }
-    }*/
+    }
   
     // TODO: XMLInfo
     //field.xml = adaptXMLInfo(prop.schema);
```

Comment:
Should this TODO remain?

Reply:
XML support hasn't been wired up yet and I don't know if tsp and/or tcgc even support it (or if they ever will).